### PR TITLE
Scala 3 minimal support

### PR DESCRIPTION
**Pull request by liucijus**
This adds minimal support for Scala 3:
- Allow to compile Scala 3 code
- Testing should work with custom toolchains
- Dependency Analyzer is an empty implementation which does not do any analysis

Also added a simple Scala 3 workspace example to validate the changes, while the rest of Rules Scala tests are not fixed to work with Scala 3.

Issue #1269 

Thanks @cocreature for the initial work

### CODE REVIEW

**Review by simuons — APPROVED** [review]

File: scala/private/macros/scala_repositories.bzl [hunk 1]
```diff
@@ -41,6 +42,28 @@ def rules_scala_setup():
...
+    "io_bazel_rules_scala_scalactic",
+    "io_bazel_rules_scala_scala_xml",
+    "io_bazel_rules_scala_scala_parser_combinators",
+] if SCALA_MAJOR_VERSION.startswith("2") else [
+    "io_bazel_rules_scala_scala_library",
+    "io_bazel_rules_scala_scala_compiler",
+    "io_bazel_rules_scala_scala_interfaces",
+    "io_bazel_rules_scala_scala_tasty_core",
+    "io_bazel_rules_scala_scala_asm",
+    #    "io_bazel_rules_scala_scala_reflect",
```
[the remark this answers is not in the document]
Comment (liucijus) [reply scala/private/macros/scala_repositories.bzl]: removed

**Comment by johnynek** [comment]
one thing that isn't clear to me is how this interacts with ijar. I don't know the role of tasty in scala 3. I think we should document how this is working.

My fear is that we are making ijars, but actually scala 3 is reading tasty files and not actually using the ijar anyway, so we are only complicating matters and hurting things. But this could be wrong.

What I think we want ideally, is for tasty to support the idea of compilation interfaces, and have a scala specific way to do something like `itasty` but I don't grok how tasty is working now, so this could be off base.

**Comment by liucijus** [comment]
ijar should keep tasty files (https://github.com/bazelbuild/bazel/commit/1489f0f4cae3e9247a70e4003ab76bef45c5b986) - I haven't verified if this works though.

**Comment by johnynek** [comment]
but if you retain the entire tasty file, doesn't that retain the entire AST? So, doesn't that kill the point of ijar for scala (i.e. any change to the implementation will still cause a downstream recompilation?)

And if any change causes a downstream recompilation, then you might as well use a transitive compilation classpath so you minimize confusing issues with symbols being missing.

**Comment by liucijus** [comment]
> but if you retain the entire tasty file, doesn't that retain the entire AST? So, doesn't that kill the point of ijar for scala (i.e. any change to the implementation will still cause a downstream recompilation?)
> And if any change causes a downstream recompilation, then you might as well use a transitive compilation classpath so you minimize confusing issues with symbols being missing.

That's correct. There's plenty of work to be done to get full support for Scala 3.

**Pull request merged**